python/sdss_access/path/path.py

```diff
@@ -716,3 +721,41 @@ def definitiondir(self, filetype, **kwargs):
 
 class AccessError(Exception):
     pass
+
+
+def _extract(template, example):
+
+    # escape the envvar $, any dots, and forward slashes
+    subtemp = template.replace('$', '\\$') \
+                      .replace('.', '\\.') \
+                      .replace('/', '\/')
+
+    # define named search pattern.
+    named_search = re.sub('{(\w+)}', '(?P<\\1>[\\w\\d+-_]+)', subtemp)
```

Comment:
Is this the correct syntax for a named capture group with a capture number, `'(?P<\\1>[\\w\\d+-_]+)'`?  Trying to run this function as is gives me the error `error: bad escape \w at position 8` here.  If I make each string in the `re.sub` raw, then the substitution works but the function fails with `error: bad character in group name '\\1' at position 6` on line `match = re.compile(named_search).search(example)`.  If I replace the `\\1` capture number with `name` , then the function returns but calls all variables the `name` reference.   I'm not familiar enough with regex named capture groups and capture group numbers to suss this out.  I'll keep playing with it though.

Reply:
like to have a package to help do `regex-alchemy` for python